packages/core/src/utils/data-attributes.ts

```diff
@@ -1,20 +1,17 @@
-type DataAttr = Record<string, string>;
+type DataAttr = { [key: string]: string };
 
 type SingleDataAttrValue = string | number | boolean | null | undefined;
-type SingleDataAttrReturn = DataAttr | Record<string, never>;
 
-export function createDataAttribute(key: string, value: SingleDataAttrValue): SingleDataAttrReturn {
-    if (value === null || value === undefined) {
-        return {};
-    }
-
-    const dataKey = `data-${key}`;
+type ReturnByValue<T> = T extends null | undefined ? {} : DataAttr;
 
-    if (typeof value === 'boolean') {
-        return value ? { [dataKey]: '' } : {};
+export function createDataAttribute<T extends SingleDataAttrValue>(
+    key: string,
+    value: T,
+): ReturnByValue<T> {
+    if (value == null || value === false) {
+        return {} as ReturnByValue<T>;
     }
-
-    return { [dataKey]: String(value) };
+    return { [`data-${key}`]: value === true ? '' : String(value) } as ReturnByValue<T>;
```

Comment:
```suggestion
        return {};
}
    return { [`data-${key}`]: value === true ? '' : String(value) };
```

`as` keyword is not needed because return type is explict.